test/test_um2netcdf.py

```diff
@@ -395,7 +403,10 @@ def __init__(self, item_code, var_name=None, attributes=None, units=None):
         self.units = None or units
         self.standard_name = None
         self.long_name = None
-        self.coord = {}
+        # TODO: Can I remove this... It breaks DummyCubeWithCoords
+        # It could be required if apply_mask is called during process
+        # tests. Would cause a KeyError anyway?
+        # self.coord = {}
```

Comment:
The `self.coord = {}` is likely my error as `coord` is a lookup function in `iris`.

There's a few options for handling this:
* delete the line entirely
* replace with a `def coords(): raise NotImplementedError("coords needs implementation...")` in the class

I'd personally add `coords()` with an error, so any future tests using the method will fail fast. As there's a few dummy cube type objects across PRs, it's likely these will need refactoring into a couple of reusable DummyCubes which work across most/all of the tests. Other tests can subclass DummyCube to specialise the fake API too.

Reply:
I've gone with the `coords()` method which raises a `NotImplementedError` when called for the base `DummyCube`